inst/templates/ad_adrs_basic.R

```diff
@@ -0,0 +1,376 @@
+adsl_vars <- exprs(RANDDT)
+
+# Join ADSL vars to RS
+adrs <- rs %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(STUDYID, USUBJID)
+  )
+
+# ---- Company-specific pre-processing ----
+
+# Filtering to select Overall Response records - here we used Investigator records
+# but all these steps could equally be repeated for Independent Review Facility
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Tumor Response",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1"
+  )
+
+# Date imputations - here we impute missing day to last possible date
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  mutate(AVISIT = VISIT)
+
+# Set numeric analysis value - here RECIST 1.1 response values are expected
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp(AVALC)
+  )
+
+# Set analysis flag to include only the records that should contribute to the
+# parameter derivations - here only valid assessments and those occurring on or
+# after randomization date, if there is more than one assessment per date the
+# worst one is flagged
+adrs <- adrs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "last"
+    ),
+    filter = !is.na(AVAL) & ADT >= RANDDT
+  )
+
+# ---- Parameter derivations ----
+
+# Progressive disease
+adrs <- adrs %>%
+  derive_extreme_records(
+    dataset_ref = adsl,
+    dataset_add = adrs,
+    by_vars = exprs(STUDYID, USUBJID),
+    filter_add = PARAMCD == "OVR" & AVALC == "PD" & ANL01FL == "Y",
+    order = exprs(ADT, RSSEQ),
+    mode = "first",
+    exist_flag = AVALC,
+    set_values_to = exprs(
+      PARAMCD = "PD",
+      PARAM = "Disease Progression by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+
+# Define the progressive disease source location
+pd <- date_source(
+  dataset_name = "adrs",
+  date = ADT,
+  filter = PARAMCD == "PD" & AVALC == "Y"
+)
+
+# Response
+adrs <- adrs %>%
+  derive_param_response(
+    dataset_adsl = adsl,
+    filter_source = PARAMCD == "OVR" & AVALC %in% c("CR", "PR") & ANL01FL == "Y",
+    source_pd = pd,
+    source_datasets = list(adrs = adrs),
+    set_values_to = exprs(
+      PARAMCD = "RSP",
+      PARAM = "Response by Investigator (confirmation not required)",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
```

Comment:
when trying to run this function derive_param_response , it giving me error.
Quitting from lines 406-422 [unnamed-chunk-19] (adrs_basic.Rmd)
Error:
! `derive_param_extreme_event()` was deprecated in admiral 0.11.0 and is
  now defunct.
i Please use `derive_extreme_records()` instead.
Backtrace:
 1. adrs %>% ...
 7. admiral::derive_param_extreme_event(...)
 8. lifecycle::deprecate_stop(...)
 9. lifecycle:::deprecate_stop0(msg)

Reply:
Did you use the latest admiralonco version?

We updated `derive_param_response()` in admiralonco 0.4.0 such that `derive_extreme_records()` is used instead of `derive_param_extreme_event()`.